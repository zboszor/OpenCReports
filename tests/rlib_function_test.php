<?php
/*
 * OpenCReports test
 * Copyright (C) 2019-2025 Zoltán Böszörményi <zboszor@gmail.com>
 * See COPYING.LGPLv3 in the toplevel directory.
 */

require_once 'test_common.php';

function my_add_func($a, $b) {
	return $a + $b;
}

function my_concat_func($s, $n) {
	return str_repeat($s, $n);
}

$o = rlib_init();
rlib_add_function($o, "my_add", "my_add_func", 2);
rlib_add_function($o, "my_concat", "my_concat_func", 2);

$e = $o->expr_parse("my_add(1,2)");
$e->optimize();
eval_print_expr($e);

$e = $o->expr_parse("my_concat('aaa', 3, 4)");
echo $o->expr_error() . PHP_EOL;

$e = $o->expr_parse("my_concat('abc', 3)");
$e->optimize();
eval_print_expr($e);
