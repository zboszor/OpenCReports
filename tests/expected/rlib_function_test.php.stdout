Expression: my_add(1.000000,2.000000)
Evaluated: (number)3.000000
invalid number of operands for my_concat at or near "my_concat('aaa', 3, 4)"
Expression: my_concat((string)abc,3.000000)
Evaluated: (string)abcabcabc
